### ISSUE TITLE
Lists within messages

### ISSUE THREAD

**Issue by ArtoszBart** [amannn/next-intl#1610]
### Is your feature request related to a problem? Please describe.

`next-intl` provides a list formatting function, such as `format.list(items, {type: 'conjunction'})`. However, it would be great if the `t()` function could accept an array as an argument, allowing you to include translated text while letting `next-intl` handle conjunctions or disjunctions automatically.

### Describe the solution you'd like

`i18next` already provides a good solution, as shown [here](https://www.i18next.com/translation-function/formatting#list).

### Describe alternatives you've considered

For example, a similar solution could look like this:

en.json
```json
{
  "list_example": "I used {listVariable, list}"
}
```

.ts
```ts
const listVariable = ['Next.js', 'Typescript', 'next-intl'];
t('list_example', { listVariable, type: 'conjunction' })
```

the output would be:
```
I used Next.js, TypeScript, and next-intl.
```

**Comment by amannn** [comment]
I agree it could be useful, but `next-intl` currently depends on `intl-messageformat` which unfortunately doesn't support this (I think because it's not spec'ed by ICU).

The alternative is:
```tsx
t('list', items: format.list(items));
```
